satellite/metainfo/validation.go

```diff
@@ -371,3 +381,29 @@ func (endpoint *Endpoint) validatePieceHash(ctx context.Context, piece *pb.Remot
 	}
 	return nil
 }
+
+func (endpoint *Endpoint) getUserAgent(ctx context.Context) (_ *uuid.UUID, err error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		if userAgent, ok := md["user-agent"]; ok {
+			if len(userAgent) == 1 {
+				ua := useragent.Parse(userAgent[0])
+				if ua == nil {
+					endpoint.log.Sugar().Info("invalid user-agent found %s", userAgent[0])
+					return nil, status.Errorf(codes.InvalidArgument, err.Error())
+				}
+				partnerID, err := rewards.GetPartnerID(ua.Name)
```

Comment:
Should we also check if the `ua.PartnerID` is equal to the `partnerID` that we get from the satellite?

Reply:
After the meeting today, we will not validate the partner ID here and only validate it during billing